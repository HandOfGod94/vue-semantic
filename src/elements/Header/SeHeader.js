import SeHeaderFunctional from './SeHeader.functional'

export default {
  name: 'se-header',
  props: {
    as: { type: String, required: true },
    inverted: Boolean,
    sub: Boolean,
    block: Boolean,
    icon: String,
    iconHeader: Boolean,
    float: String,
    disabled: Boolean,
    align: String,
    justified: Boolean,
    color: String,
    content: String,
    subheader: String
  },

  computed: {
    classObject () {
      return {
        ui: true,
        inverted: this.inverted,
        [this.color]: this.color,
        [`${this.float} floated`]: this.float,
        [`${this.align} aligned`]: this.align,
        disabled: this.disabled,
        icon: this.iconHeader,
        sub: this.sub,
        block: this.block,
        justified: this.justified,
        header: true
      }
    }
  },

  render (create) {
    return create(SeHeaderFunctional, {
      attrs: {
        ...this.$props,
        slots: this.$slots
      },
      class: this.classObject
    })
  }
}
