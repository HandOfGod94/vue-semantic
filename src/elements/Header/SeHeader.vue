<template>
  <component :is="as" :class="classObject">
    <i v-if="icon" :class="`${icon} icon`"></i>
    <slot></slot>
    <div v-if="hasContent" class="content">
      <slot name="content">
        {{content}}
      </slot>
      <div v-if="hasSubheader" class="sub header" >
        <slot name="subheader">
          {{subheader}}
        </slot>
      </div>
   </div>
  </component>
</template>

<script>
export default {
  name: 'se-header',
  props: {
    as: { type: String, required: true },
    inverted: Boolean,
    sub: Boolean,
    block: Boolean,
    iconHeader: Boolean,
    icon: String,
    disabled: Boolean,
    align: String,
    justified: Boolean,
    color: String,
    content: String,
    subheader: String
  },
  computed: {
    classObject () {
      return {
        ui: true,
        inverted: this.inverted,
        [this.color]: this.color,
        [`${this.align} aligned`]: this.align,
        disabled: this.disabled,
        icon: this.iconHeader,
        sub: this.sub,
        block: this.block,
        justified: this.justified,
        header: true
      }
    },
    hasContent () { return !!this.$slots['content'] || this.content },
    hasSubheader () { return !!this.$slots['subheader'] || this.subheader }
  }
}
</script>

<style>

</style>
