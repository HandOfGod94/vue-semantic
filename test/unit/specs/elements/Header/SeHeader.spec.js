import SeHeader from '@/elements/Header/SeHeader'
import { mount } from '@vue/test-utils'

const headerContent = 'Hello World'

describe('SeHeader basic snapshot test', () => {
  it('should match with exisiting snapshot', () => {
    const data = {propsData: {as: 'h1'}, slots: {default: headerContent}}
    const wrapper = mount(SeHeader, data)
    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('SeHeader classes test', () => {
  it('should have "disabled" class', () => {
    const data = {propsData: {as: 'h1', disabled: true}, slots: {default: headerContent}}
    const wrapper = mount(SeHeader, data)
    expect(wrapper.classes()).toContain('disabled')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have "img" as child', () => {
    const data = {propsData: {as: 'h1', image: 'static/user.png'}, slots: {default: headerContent}}
    const wrapper = mount(SeHeader, data)
    expect(wrapper.contains('img.ui.image')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have "icon" child', () => {
    const data = {propsData: {as: 'h1', icon: 'settings'}, slots: {default: headerContent}}
    const wrapper = mount(SeHeader, data)
    expect(wrapper.contains('i')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have "div" child as content', () => {
    const data = {propsData: {as: 'h1', icon: 'settings', content: 'Hello World'}}
    const wrapper = mount(SeHeader, data)
    expect(wrapper.contains('i.settings.icon')).toBe(true)
    expect(wrapper.contains('div.content')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have "block" class', () => {
    const data = {propsData: {as: 'h1', block: true}, slots: {default: headerContent}}
    const wrapper = mount(SeHeader, data)
    expect(wrapper.classes()).toContain('block')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have aligned class', () => {
    const data = {propsData: {as: 'h1', align: 'right'}, slots: {default: headerContent}}
    const wrapper = mount(SeHeader, data)
    expect(wrapper.contains('.right.aligned')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have "justified" class', () => {
    const data = {propsData: {as: 'h1', justified: true}, slots: {default: headerContent}}
    const wrapper = mount(SeHeader, data)
    expect(wrapper.classes()).toContain('justified')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have color class', () => {
    const data = {propsData: {as: 'h1', color: 'orange'}, slots: {default: headerContent}}
    const wrapper = mount(SeHeader, data)
    expect(wrapper.classes()).toContain('orange')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
