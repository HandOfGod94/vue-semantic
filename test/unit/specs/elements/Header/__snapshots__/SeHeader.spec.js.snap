// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeHeader basic snapshot test should match with exisiting snapshot 1`] = `<h1 class="ui header">Hello World</h1>`;

exports[`SeHeader classes test should have "block" class 1`] = `<h1 class="ui block header">Hello World</h1>`;

exports[`SeHeader classes test should have "disabled" class 1`] = `<h1 class="ui disabled header">Hello World</h1>`;

exports[`SeHeader classes test should have "icon" child 1`] = `
<h1 class="ui icon header">
    <i class="settings icon"></i>Hello World</h1>
`;

exports[`SeHeader classes test should have "justified" class 1`] = `<h1 class="ui justified header">Hello World</h1>`;

exports[`SeHeader classes test should have aligned class 1`] = `<h1 class="ui right aligned header">Hello World</h1>`;

exports[`SeHeader classes test should have appropriate structure for content and subheader 1`] = `
<h1 class="ui header">
    <div class="content">Hello World
        <div class="sub header">Hello World Subheader</div>
    </div>
</h1>
`;

exports[`SeHeader classes test should have appropriate structure when slots are used 1`] = `
<h1 class="ui header">
    <div class="content">Hello World
        <div class="sub header">Hello World Subheader</div>
    </div>
</h1>
`;

exports[`SeHeader classes test should have color class 1`] = `<h1 class="ui orange header">Hello World</h1>`;
